prow/cmd/crier/main.go

```diff
@@ -182,10 +182,7 @@ func main() {
 		if cfg().SlackReporter == nil {
 			logrus.Fatal("slackreporter is enabled but has no config")
 		}
-		slackConfig := func() *config.SlackReporter {
-			return cfg().SlackReporter
-		}
-		slackReporter, err := slackreporter.New(slackConfig, o.dryrun, o.slackTokenFile)
+		slackReporter, err := slackreporter.New(cfg().SlackReporterConfigs.GetSlackReporter, o.dryrun, o.slackTokenFile)
```

Comment:
Please pass in `cfg` or a subset of it as a closure that is resolved on-demand, otherwise the Slack reporter won't pick up config changes without a resart

Reply:
Now passing in a closure to capture config updates:
```golang
slackConfig := func(refs *prowapi.Refs) (*config.SlackReporter, error) {
	return cfg().SlackReporterConfigs.GetSlackReporter(refs)
}
```